### ISSUE TITLE
API using a blocking function - Warning

### ISSUE THREAD

**Issue by gigatexel** [JoDehli/PyLoxone#277]
### Describe the bug

The API (https://github.com/JoDehli/PyLoxone/blob/master/custom_components/loxone/api.py#L803) is using a file-open function that is blocking the event loop. This is causing HA to throw warnings:

```
2024-06-13 08:47:26.738 WARNING (MainThread) [homeassistant.util.loop] Detected blocking call to open inside the event loop by custom integration 'loxone' at custom_components/loxone/api.py, line 803: with open(persist_token, "w") as write_file: (offender: /config/custom_components/loxone/api.py, line 803: with open(persist_token, "w") as write_file:), please create a bug report at https://github.com/JoDehli/PyLoxone/issues
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/usr/src/homeassistant/homeassistant/__main__.py", line 223, in <module>
    sys.exit(main())
  File "/usr/src/homeassistant/homeassistant/__main__.py", line 209, in main
    exit_code = runner.run(runtime_conf)
  File "/usr/src/homeassistant/homeassistant/runner.py", line 190, in run
    return loop.run_until_complete(setup_and_run_hass(runtime_config))
  File "/usr/local/lib/python3.12/asyncio/base_events.py", line 672, in run_until_complete
    self.run_forever()
  File "/usr/local/lib/python3.12/asyncio/base_events.py", line 639, in run_forever
    self._run_once()
  File "/usr/local/lib/python3.12/asyncio/base_events.py", line 1988, in _run_once
    handle._run()
  File "/usr/local/lib/python3.12/asyncio/events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 742, in async_setup_locked
    await self.async_setup(hass, integration=integration)
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 594, in async_setup
    result = await component.async_setup_entry(hass, self)
  File "/config/custom_components/loxone/__init__.py", line 379, in async_setup_entry
    res = await miniserver.start_ws()
  File "/config/custom_components/loxone/miniserver.py", line 120, in start_ws
    res = await self.api.async_init()
  File "/config/custom_components/loxone/api.py", line 439, in async_init
    res = await self.acquire_token()
  File "/config/custom_components/loxone/api.py", line 740, in acquire_token
    if self.save_token() == ERROR_VALUE:
  File "/config/custom_components/loxone/api.py", line 803, in save_token
    with open(persist_token, "w") as write_file:
```

Possible fixes:
- https://community.home-assistant.io/t/blocking-call-inside-event-loop/575796
- https://github.com/Tinche/aiofiles

### Firmware of your Miniserver

not relevant

### HomeAssistant install method

not relevant

### Version of HomeAssistant

2024.6.2

### Version of Pyloxone

Latest

### Update pyloxone

Yes

### Log

Settings for the log can be found on the main page under the section Log Configuration

**Comment by gigatexel** [comment]
It's a warning, but it actually fills up the home-assistant.log file.

I think there are more integrations that are having the same issue, more dan 30k log lines with simmilar warnings.

**Comment by gigatexel** [comment]
lol :-)

https://github.com/home-assistant/core/issues?q=frozen+runpy+label%3A%22blocking+event+loop%22

**Comment by gigatexel** [comment]
Fixed!

**Comment by Petahh** [comment]
Hello @JoDehli, even though I'm on the 0.6.10 version which should contain the fix I'm still getting the error:

`Detected blocking call to open inside the event loop by custom integration 'loxone' at custom_components/loxone/api.py, line 936: from Crypto.PublicKey import RSA (offender: /usr/local/lib/python3.12/ctypes/util.py, line 276: with open(filepath, 'rb') as fh:), please create a bug report at https://github.com/JoDehli/PyLoxone/issues Traceback (most recent call last): File "<frozen runpy>", line 198, in _run_module_as_main File "<frozen runpy>", line 88, in _run_code File "/usr/src/homeassistant/homeassistant/__main__.py", line 223, in <module> sys.exit(main()) File "/usr/src/homeassistant/homeassistant/__main__.py", line 209, in main exit_code = runner.run(runtime_conf) File "/usr/src/homeassistant/homeassistant/runner.py", line 190, in run return loop.run_until_complete(setup_and_run_hass(runtime_config)) File "/usr/local/lib/python3.12/asyncio/base_events.py", line 672, in run_until_complete self.run_forever() File "/usr/local/lib/python3.12/asyncio/base_events.py", line 639, in run_forever self._run_once() File "/usr/local/lib/python3.12/asyncio/base_events.py", line 1988, in _run_once handle._run() File "/usr/local/lib/python3.12/asyncio/events.py", line 88, in _run self._context.run(self._callback, *self._args) File "/usr/src/homeassistant/homeassistant/config_entries.py", line 742, in async_setup_locked await self.async_setup(hass, integration=integration) File "/usr/src/homeassistant/homeassistant/config_entries.py", line 594, in async_setup result = await component.async_setup_entry(hass, self) File "/config/custom_components/loxone/__init__.py", line 379, in async_setup_entry res = await miniserver.start_ws() File "/config/custom_components/loxone/miniserver.py", line 120, in start_ws res = await self.api.async_init() File "/config/custom_components/loxone/api.py", line 408, in async_init rsa_gen = self.init_rsa_cipher() File "/config/custom_components/loxone/api.py", line 936, in init_rsa_cipher from Crypto.PublicKey import RSA`

**Comment by JoDehli** [comment]
@Petahh I tested it against the newest dev branch and the latest release of home assistant. I have no messages like you since I fixed it. 

The error comes from a "with open(filepath, 'rb') as fh:)" statement which is in a system library not in pyloxone. 
I think the Homeassistant guys are overdoing it a bit with their alerts and optimisations. In the meantime, it's no longer any fun with Homeassistant.
